### ISSUE TITLE
Laptop randomly hangs

### ISSUE THREAD

**Issue by ichazz** [jaromeyer/XPS9570-Catalina#21]
I have went through your guide a couple of times now (including the newly updated EFI from a few days ago. I am finding that the laptop will randomly hang and i can't find a pattern, it seems utterly random. Every part of laptop seems to be working perfectly otherwise.

Do you think this could be a thermal issue? I haven't undervolted the CPU or repasted.

Another possibility is I installed a 2tb SSD (WD blue), could it be overheating/incorrectly configured? 

The laptop is under no stress when these hangs occur. Either iddle installing/copying files.

**Comment by frbuccoliero** [comment]
I’d exclude the ssd since me and @jaromeyer  have both the wd blue (512 gb and 1tb) 

Is the pc freezing or just slowing down?

**Comment by ichazz** [comment]
Ok, that's great news re the SSD.
The PC is completely freezing. The screen stays active but the mouse/keyboard stop responding. I have checked to see if the print screen has disabled one or other, but nothing.

**Comment by ichazz** [comment]
Sorry, i totally forgot to mention that I have installed Mojave because I need to use Wine from time to time. I assumed this wouldn't be an issue since the computer boots and seems fast and responsive. But no stone left unturned.

**Comment by frbuccoliero** [comment]
Mh.. never tried mojave. I fresh installed catalina since i got the pc. Why don't you double boot windows so you don't need wine anymore and you can install catalina. Or also a virtual machine might do the job

**Comment by jaromeyer** [comment]
I'm pretty sure it's not a thermal issue. Your SSD is connected over SATA, not NVMe, right? What exactly happens then it crashes? Does it freeze forever or do you get some sort of debug output? If its a kernel panic, there should be logs somewhere. Do you get that Apple popup telling something went wrong once you boot again? And I've never tried Mojave with OC. Like @frbuccoliero you could update to Catalina and use a VM.

**Comment by ichazz** [comment]
So sorry for my ignorance but how do you mean SATA and not NVMe? I replaced the stock m.2 drive with a 2tb m.2 and i'm afraid that's all i know.

For the freezing, it locks entirely and is completely unresponsive. When i reboot it boots as if nothing happened, no apple message about a crash. 

For mojave with OC i have two other hack desktops running it very happily (I'm on one right now in fact).

**Comment by ichazz** [comment]
I did a fresh install with Catalina and everything was going fine as always but after about an hour the freeze occured.

**Comment by jaromeyer** [comment]
under system report, you can see if your drive appears under SATA/SATA Express or NVMExpress. also have you tried generating your own CPUFriendDataProvider.kext? What WiFi card are you using? Try disabling both WiFi and BT in the BIOS

**Comment by ichazz** [comment]
System report says there is no NVMe device and the drive shows up under the SATA section.
I have also tried dissabling the SD card reader-the hang still occured.
I have not tried CPUFriendDataProvider.kext as I couldn't find a clear enough guide to confidently use it. I will look again and report back.
I am using a DW1830. After disabling both wifi+bluetooth the computer seems to work fine. I then activated wifi (not BT) and the computer crashed again after 15 mins.
